assets/js/newsletter.js

```diff
@@ -0,0 +1,15 @@
+---
+# Empty front matter to enable Jekyll processing
+---
+$("#newsletter-modal").on('shown.bs.modal', function(w, d, s, o, f, js, fjs) {
+    w['ecm-widget'] = o;
+    w[o] = w[o] || function() {
+        (w[o].q = w[o].q || []).push(arguments)
+    };
+    js = d.createElement(s), fjs = d.getElementsByTagName(s)[0];
+    js.id = '{{ site.newsletter }}';
```

Comment:
I'd prefer not to have the JS preprocessed by Jekyll/Liquid. It's not very maintainable in the long run, especially if we're going to want to test the scripts. It would be better to put the necessary variables as globals in the generated HTML.

Reply:
Fair enough, though currently above my skills. How do I set a global variable in HTML so that the JS can read it?

As a second issue: Do we not want to run JS tests only _after_ Jekyll build anyway? For example to test search, Jekyll build is essential (it generated the search JSON).

Reply:
I agree with this. At the same time, we have a precedent in search.json. IMO, that can be easily addressed together in a separate PR, focused on unit-tests...

Reply:
@jankrcal, I'm not sure to what "agree with this" refers :-). Do you prefer NOT to have Jekyll pre-process JS, or keep it as suggested now and potentially change in a separate PR when adding tests?

Reply:
I've resolved this for now.